### PR TITLE
PWX-32732 : use portworx-restricted based on IsPrivileged flag

### PR DESCRIPTION
**Pull request by nikita-bhatia**
**What this PR does / why we need it**:
Do not use portworx-restricted SCC by default, use the old way and allow to enable/disable it via annotation

**Which issue(s) this PR fixes** (optional)
Closes # https://portworx.atlassian.net/browse/PWX-32732

### CODE REVIEW

File: drivers/storage/portworx/component/csi.go [hunk 1]
```diff
@@ -332,6 +326,18 @@ func (c *csi) createClusterRole(
...
+		clusterRole.Rules = append(
+			clusterRole.Rules,
+			rbacv1.PolicyRule{
+				APIGroups:     []string{"security.openshift.io"},
+				Resources:     []string{"securitycontextconstraints"},
+				ResourceNames: []string{PxRestrictedSCCName},
+				Verbs:         []string{"use"},
+			},
+		)
+	}
```
Comment (zoxpx) [inline drivers/storage/portworx/component/csi.go:339]: This does not look right -- shouldn't it be something like:

```go
        sccName := PxSCCName
	if !pxutil.IsPrivileged(cluster) {
            sccName := PxRestrictedSCCName
        }
	clusterRole.Rules = append(
		clusterRole.Rules,
		rbacv1.PolicyRule{
			APIGroups:     []string{"security.openshift.io"},
			Resources:     []string{"securitycontextconstraints"},
			ResourceNames: []string{sccName},
			Verbs:         []string{"use"},
		},
	)
```

File: drivers/storage/portworx/testspec/csiClusterRole_k8s_1.14.yaml [hunk 2]
```diff
@@ -54,10 +54,6 @@ rules:
 - apiGroups: ["coordination.k8s.io"]
   resources: ["leases"]
   verbs: ["*"]
-- apiGroups: ["security.openshift.io"]
-  resources: ["securitycontextconstraints"]
-  resourceNames: ["portworx-restricted"]
-  verbs: ["use"]
```
Comment (zoxpx) [inline drivers/storage/portworx/testspec/csiClusterRole_k8s_1.14.yaml:60]: same here...

**Review by zoxpx — CHANGES_REQUESTED** [review]
I think this PR will require some more work (see below)

File: drivers/storage/portworx/component/csi.go (hunk 1 above)
Comment (nikita-bhatia) [reply drivers/storage/portworx/component/csi.go]: done

File: drivers/storage/portworx/testspec/csiClusterRole_k8s_1.11.yaml [hunk 3]
```diff
@@ -51,10 +51,6 @@ rules:
 - apiGroups: ["coordination.k8s.io"]
   resources: ["leases"]
   verbs: ["*"]
-- apiGroups: ["security.openshift.io"]
-  resources: ["securitycontextconstraints"]
-  resourceNames: ["portworx-restricted"]
-  verbs: ["use"]
 - apiGroups: ["policy"]
```
[the remark this answers is not in the document]
Comment (nikita-bhatia) [reply drivers/storage/portworx/testspec/csiClusterRole_k8s_1.11.yaml:58]: changed back to default SCCname, also added UTs with isPrivileged anotation "false"

File: drivers/storage/portworx/component/lighthouse.go [hunk 4]
```diff
@@ -147,75 +147,79 @@ func (c *lighthouse) createServiceAccount(
...
+				APIGroups:     []string{"policy"},
+				Resources:     []string{"podsecuritypolicies"},
+				ResourceNames: []string{constants.PrivilegedPSPName},
+				Verbs:         []string{"use"},
 			},
 		},
-	)
+	}
+
+	return k8sutil.CreateOrUpdateClusterRole(c.k8sClient, clusterRole)
```
Comment (zoxpx) [inline drivers/storage/portworx/component/lighthouse.go:222]: This seems to be a new addition... was this a mistake, or was this line required ?

**Review by zoxpx — APPROVED** [review]
This looks good -- thanks Nikita  (one question logged below)

File: drivers/storage/portworx/component/lighthouse.go (hunk 4 above)
Comment (nikita-bhatia) [reply drivers/storage/portworx/component/lighthouse.go:222]: This method was returning **k8sutil.CreateOrUpdateClusterRole** , changed it to last line to add conditional statement of SCC name before creating the cluster role.

**Pull request merged**